src/history_tree_node.rs

```diff
@@ -141,6 +141,25 @@ impl HistoryTreeNode {
         }
     }
 
+    pub(crate) async fn batch_get_from_storage<S: V2Storage + Send + Sync>(
+        storage: &S,
+        keys: Vec<NodeKey>,
```

Comment:
I'll refactor this later, but maybe I'll borrow the keys as well rather than clone them. It'll require a change on the storage trait (todo for me :p )